tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,61 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   ASSERT_TRUE(simple_wildcard_match("test string", "test string"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "test string*"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "tes? string"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "test *"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "test strin1"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "te*t"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "test string1"));
+}
+
+TEST(set_traced_syscalls_int, basic)
+{
+   tracing_mock mock;
+   const char too_long[TRACED_SYSCALLS_STR_LEN] = {'a'};
+   memset((void *) too_long, 'a', TRACED_SYSCALLS_STR_LEN);
+   const char almost_too_long[TRACED_SYSCALLS_STR_LEN-1] = {'a'};
+   memset((void *) almost_too_long, 'a', TRACED_SYSCALLS_STR_LEN-1);
+   ASSERT_EQ(set_traced_syscalls_int(too_long), -ENAMETOOLONG);
+   ASSERT_EQ(set_traced_syscalls_int(almost_too_long), -ENAMETOOLONG);
+
+   EXPECT_CALL(mock, handle_sys_trace_arg(_))
+      .WillOnce(Return(1));
+   ASSERT_EQ(set_traced_syscalls_int(",,,"), 1);
+
+   EXPECT_CALL(mock, handle_sys_trace_arg(_))
+      .WillOnce(Return(1));
+   ASSERT_EQ(set_traced_syscalls_int("abc"), 1);
+
+   bool traced_syscalls_mock[MAX_SYSCALLS] = {0};
+   traced_syscalls = traced_syscalls_mock;
+   traced_syscalls[0] = 1;
+   char single_char = 'a';
+   traced_syscalls_str = &single_char;
```

Comment:
This still doesn't make any sense. I had to check the code and try to figure out myself what you're trying to achieve. "Not crashing" is not enough. The buffer is supposed to be `TRACED_SYSCALLS_STR_LEN` long. Just re-use the `tracing_str` buffer above. This way, you'll have a single buffer for all the purposes.